src/main/java/com/google/sps/data/BadgeData.java

```diff
@@ -0,0 +1,38 @@
+package com.google.sps.data;
+
+import java.util.HashMap;
+
+public final class BadgeData {
+  public static final String GARD_BADG_1 = "GARD_BADG_1";
+  public static final String RECY_BADG_1 = "RECY_BADG_1";
+  public static final String WAST_BADG_1 = "WAST_BADG_1";
+
+  public static final String GARD_BADG_3 = "GARD_BADG_3";
+  public static final String RECY_BADG_3 = "RECY_BADG_3";
+  public static final String WAST_BADG_3 = "WAST_BADG_3";
+
+  private static final String YOU_COMPLETE_ONE = "You've Completed 1 ";
+  private static final String YOU_COMPLETE_THREE =  "You've Completed 3 ";
+
+  private static final String GARDENING_CHALLENGE = "Gardening Challenge";
+  private static final String RECYCLE_CHALLENGE = "Recycle Challenge";
+  private static final String WASTE_CHALLENGE = "Waste Challenge";
+
+  private static final int COMPL_ONE_CHAL = 1;
+  private static final int COMPL_THREE_CHAL = 3;
+  private static final int COMPL_FIVE_CHAL = 5;
+
+  public static HashMap<String, Badge> BADGE_MAP = createMap();
+
+  private static HashMap<String, Badge> createMap() {
+    HashMap<String, Badge> badge_map = new HashMap<String, Badge>();
```

Comment:
A bit confused here -- are you just manually creating a sample of badges? Maybe you want to get all the challenges from the challenges servlet and create some sort of function/iteration to fill the badge map instead of adding manually?

Reply:
I also am not sure if this much data needs to be stored for each individual challenge, it works as is but alternatively could just store the number of each type of challenge that the user has completed if the badges are of the format "You've completed X Gardening Challenges" and then fill in the variables on the frontend? I might just be missing something about how you are doing this

Reply:
Well the BADGE_MAP is a master list like CHALLENGES_MAP in ChallengeData.Java 
Also i want to keep as much of the logic as I can to the backend

Reply:
Right, what I meant for the first comment was that every time you add a challenge in ChallengeData.java you will have to manually add its corresponding badge so I was wondering if there was a way to automate/connect those two pieces more fluidly.

Reply:
badges don't necessarily correspond to a challenge but rather a challenge type. So as long as new challenges are added in existing types, badges won't have to change much at all.

Reply:
There's a good point here: repetitive data is usually a bad code smell. The string literals "You've Completed 1 " "You've Completed 3" are repeated here and in the JS files. One way to work around this is to transfer all badge data (description string and all) in the doGet in '/badges'. The JS would use the data provided by GET /badges to display the user visible text.
 By storing all string literals in your backend, you only have a single location to modify if you want to change the text around.

Moreover, you can also include the resource location (e.g. "resources/recy_medal_1.png") in the backend badge data and transfer it to the frontend during the GET /badges request.

Reply:
the "You've Completed" lines were meant to be deleted.  I had them in the front end to for testing. I planned to have all the badge data in the backend but I didn't get around to fully transferring the information.